tici/hardware.py: improve exception handling with decorators

deanlee opened the pull request:
The get_modem() method may fail and raise exceptions. encapsulating it within a try-except block ensures safe access to the modem.

deanlee committed:
improve error handling

adeebshihadeh reviewed (changes requested):
This is generally a problem with the Hardware classes. We have to come up with a better pattern than the try/excepts everywhere.

deanlee committed:
use decorator to handle exception

TODO

deanlee commented:
Most functions use decorators to handle exceptions now, except for `get_nvme_temperatures`, `get_modem_data_usage`, and `get_imei`. This is because both `get_nvme_temperatures` and `get_modem_data_usage` will throw exceptions if `wwan0` does not exist or if the command `sudo smartctl -aj /dev/nvme0` fails. If decorators were used to handle these exceptions, each call  would output messages like:
> "Exception in get_modem_data_usage: org.freedesktop.NetworkManager.UnknownDevice: No device found for the requested iface."
> "Exception in get_nvme_temperatures: Command 'sudo smartctl -aj /dev/nvme0' returned non-zero exit status 2."

`get_imei` has its exceptions handled by the caller (registration.py), so it does not need a decorator.:

> 
>       try:
>         imei1, imei2 = HARDWARE.get_imei(0), HARDWARE.get_imei(1)
>       except Exception:
>         cloudlog.exception("Error getting imei, trying again...")
>         time.sleep(1)

deanlee committed:
uploader.py: ensure proper resource management with io.BytesIO (#33108)

* ensure proper resource management with io.BytesIO

* improve

andiradulescu committed:
update mac_setup.sh brew install (#33131)

deanlee committed:
replay: add support for decompressing ZST log files (#32910)

* Add Support for Decompressing ZST Log Files

* 2 space and check magic number

* match BZ2

---------

Co-authored-by: Shane Smiskol <shane@smiskol.com>

adeebshihadeh committed:
qlog: remove mag and less CAN

fredyshox committed:
process_replay: regenerate locationd logs (#33134)

* Filter out non-relevant services in most_messages_valid check

* Set as comprehension

* Update segments

* Update ref commit

* New check for hyundai2

fredyshox committed:
paramsd: use livePose (#33099)

* Use livePose instead  of llk

* Update process replay sockets

* Fix import

* Fix calib

* Fix field name

* Dont store device_from_calib

* Update ref commit

sshane committed:
qlog_size.py: use zstd (#33137)

qlog_size: use zstd

fredyshox committed:
torqued: use livePose (#33136)

* Use livePose

* Replace it in process replay

* Add liveCalibration to messages

* Update ref commit

sshane committed:
process replay: use zst (#33138)

* test

* update refs to zst

* update

andiradulescu committed:
fix multiprocessing issue with can_replay.py on macOS (#33139)

fix multiprocessing issue with can_replay.py on mac

adeebshihadeh committed:
add roadmap to readme

maxime-desroches committed:
unpin numpy (#33132)

* unlock numpy

* remove old cache

* sheel

* lock

* remove temp

* depends

* export

* put this back

* use __version__

* move this

---------

Co-authored-by: quebec <quebec@quebec.attlocal.net>

maxime-desroches committed:
docs: fix README.md (#33142)

* md

* Update README.md

* rename

* exclude

* Update README.md

sshane committed:
test_onroad: lower lower log size bound

sshane committed:
cars: remove some external imports (#33133)

* ford and gm

* clean that up

* also this

* honda

* temp fix

* move into selfdrive.car

* clean up

* more

adeebshihadeh committed:
CI: increase shm size to 2G

sshane committed:
Ford: add long controls todo

sshane committed:
tester present msgs: use helper (#33147)

use helper

sshane committed:
deprecate busTime (#33146)

* deprecate busTime

* bump

* do car can + more

* forgot some

* bump opendbc

* fix that too

* bump

sshane committed:
bump

sshane committed:
Revert "bump"

This reverts commit 7824074aae74ff12bd56880a690234b582535748.

sshane committed:
CarController: call to super's init (#33148)

* this should be common

* super!

* bump

* and frame is common

* bump

deanlee committed:
update to new opendbc API (#32009)

* use new opendbc api

* export pandad_python

* merge master

* merge master

* bump opendbc

* bump opendbc

* improve func

* keep interface unchanged

* fix test_car_interfaces

* bump opendbc

* bump opendbc

* fix test_models

* the interface now has to convert from can capnp to list, so we should include this time

* goes from ~210 to ~240 mean ms real time

* remoe busTime

* lowercase sendcan

* consistent msgtype

* bump

* bump

* not used in lat_mpc

* space

* bump to master

---------

Co-authored-by: Shane Smiskol <shane@smiskol.com>

sshane committed:
Revert "update to new opendbc API (#32009)"

This reverts commit 0dddc97dca8bf4e023b29c35cb92fd3dd0a9a3a2.

maxime-desroches committed:
setup.sh: retry + timing (#33145)

* improve

* update readme

* stdin

* better

* allow non interactive

* specify dir

maxime-desroches committed:
Remove hexdump package (#33155)

* remove hexdump

* uv lock

* cleanup

sshane committed:
get_fw_versions: test the full stack (#33156)

* add the test

fix

* bump

* not sure why mypy didn't catch this

sshane committed:
Reapply "update to new opendbc API (#32009)" (#33151)

This reverts commit ac130001cc83734118524f59a83a96a2065dd56e.

sshane committed:
ensure alerts_generated.h is built (#33157)

don't build alerts_generated.h in test, and ensure scons builds it more reliably + clean up support

ZwX1616 committed:
camerad: use os binning (#33161)

* all

* right sz

* what is this

* Revert "what is this"

This reverts commit 3ff771926f109452a055dfb61cee06b34e498164.

* ok

* big

* update intrs

* fl

* more specifuc

* fish

* revert

* revert

---------

Co-authored-by: Comma Device <device@comma.ai>

maxime-desroches committed:
op.sh: call directly instead of sourcing (#33160)

no source

maxime-desroches committed:
setup.sh: work in non interactive mode (#33162)

* work without terminal

* failure message

* no if

maxime-desroches committed:
op.sh: fix python version check (#33163)

fix check

sshane committed:
opendbc is a package (#33159)

* bump

* rename to opendbc_repo

add symlink

* bump

* fix cabana

fix cabana

* bump

* fix whitelist paths

bump

* bump to master

* remove trailing slash

maxime-desroches committed:
op.sh: check for valid invocation (#33166)

no source

adeebshihadeh committed:
disable test_rotation until it's fixed

bongbui321 committed:
CI: fix recent test_rotation skip (#33167)

maxime-desroches committed:
ci: create ui report only on PR (#33168)

only on pr

adeebshihadeh committed:
remove pre-commit (explorations) (#33172)

* remove pre-commit (explorations)

* lint.sh

* remove rest

* add to release files

* no source

* 1min is all you need

deanlee committed:
loggerd: Initialize atomic variables to zero (#33130)

Initialize atomic variables in LoggerdState to zero

deanlee committed:
camerad: encapsulating event management within the ImgProc Class (#33119)

Encapsulating Event Management Within the ImgProc Class

deanlee committed:
params: wrap fsync and close in HANDLE_EINTR for robustness (#33154)

Wrap fsync and close in HANDLE_EINTR for Robustness

deanlee committed:
camerad: move `do_exit` to camera_qcom2.cc (#33086)

move do_exit to camera_qcom2.cc

adeebshihadeh committed:
Update stale.yaml

deanlee committed:
camerad: update frame msg related variable types to uint64_t for consistency (#33113)

Update frame_msg related variable types to uint64_t for  type safety

deanlee committed:
modeld: remove redundant pred_mu assignment (#33152)

remove redundant pred_mu assignment

deanlee committed:
uploader.py: simplify clear_locks by using rglob (#32974)

simplify clear_locks

adeebshihadeh committed:
Revert "uploader.py: simplify clear_locks by using rglob (#32974)"

This reverts commit cafca4f89176855585835ff7c67dea5a9b08ca23.

deanlee committed:
camerad: encapsulate cl_command_queue in ImgProc (#33178)

move cl queue to imgproc

adeebshihadeh committed:
rerun: misc fixups

ugtthis committed:
Docs: curl command updated (#33180)

adeebshihadeh committed:
op: simpler examples

adeebshihadeh committed:
bounties.md moved to the project README

adeebshihadeh committed:
back to pypi casadi (#33182)

commaci-public committed:
[bot] Update Python packages (#33185)

pin numpy :(

Co-authored-by: Maxime Desroches <desroches.maxime@gmail.com>

commaci-public committed:
[bot] Bump submodules (#33184)

bump submodules

Co-authored-by: Vehicle Researcher <user@comma.ai>

sshane committed:
separate car logger forwarded to cloudlog (#33170)

* car logger

* try this

* fix double printing!

* clean up

* need to import where we actually use it (card won't be in submodule)

* NullHandler does not forward everything properly, need propagate=False to stop root logger from printing

* card is staying

* set up

* forward

sshane committed:
[bot] Fingerprints: add missing FW versions from new users (#33191)

Export fingerprints

maxime-desroches committed:
setup.sh: metrics (#33175)

* save

* add sentry

* clean

* real sentry

* default

* true

* remote this

maxime-desroches committed:
fix all codespell issues (#33194)

* bring back

* remove 2020 dictionary

adeebshihadeh committed:
Revert "Remove `setproctitle` (#32716)" (#33195)

* Revert "Remove `setproctitle` (#32716)"

This reverts commit 83ac80c6c899dbf4b5b2d422947b02b57d83bccd.

* uv lock

* old name

sshane committed:
remove more car import exceptions (#33193)

* remove another exception (remove mention of events from car stack)

* guess this should've been here all along

* reorganize exceptions

* fix

sshane committed:
remove unnecessary debugging function from car_helpers.py (#33199)

* format

* better better

* huh

sshane committed:
remove CarVin parameter

sshane committed:
fix build

sshane committed:
selfdrive/car: OBD callback (#33200)

* finish

* pass empty cb

* last fix!

* i keep messing this up

* even more coverage!

* not needed

sshane committed:
fingerprinted: log platform string

This does get safely converted to a string in the logs, but in the console it's the repr

sshane committed:
selfdrive/car: ban params (#33198)

* ban params too with a callback

* all sorts of messed up

* use cloudlog

* consistent order

* order

* better type hint

* format

* this is a bit nicer

* hmm

* fix PLR1704

* no carvin

* fix process replay

sshane committed:
get_car: returns just CI (#33204)

just return CI

Commit:
tici: update spinner and text

adeebshihadeh committed:
macOS: remove devcontainer recommendation

sshane committed:
add back lint-imports (#33202)

add back

bongbui321 committed:
tools/rerun: new image API for rerun 0.18 (#33186)

* change api for video

* install prerelease 0.18

sshane committed:
selfdrive/car: ban selfdrive/controls (#33211)

ban controls (VM isn't used anymore)

sshane committed:
add exception for body

adeebshihadeh committed:
docs: add #current-projects link

adeebshihadeh committed:
sensord: error -> warning for expected failures

deanlee committed:
pandad: assemble capnp message using separate functions (#33189)

Assemble capnp message using separate functions

deanlee committed:
pandad: refactor to consolidate threads, keep only one can_send thread (#32680)

* single thread

improve comment

* Keep can_send() running in a separate thread

* send send_peripheral_state in pandad_run

* new PandaSafety class

sshane committed:
selfdrive/car: ban common (#33210)

* ban all of common & copy numpy_fast

* add numpy_fast

* these are okay

* and ban controls

* better name

* Conversions

* do utils, kalman

* clean up

* sorting

* don't forget

sshane committed:
selfdrive/car: ban the rest of the external imports (#33214)

more bans

maxime-desroches committed:
pre-commit: bring back some checks (#33196)

* add this

* add this

* nice ...

* 120

* remove

* no remote

* 0

* this

* no submodules

* like this

* try this

* validate

* like this

* split

* pass files

* no par

* check exist

* no build

* fix

* 24.04

* default

* test failure

* continue

* actually fail

* clean

* import

* fix

* line

* mypy

* clean

* simpler

* final

* better

* final clean

* not true

maxime-desroches committed:
Revert "pre-commit: bring back some checks (#33196)" (#33216)

This reverts commit 30467b44cd82973576c8073136dede6aa2fa60f0.

sshane committed:
remove CAN capnp conversion from car interfaces (#33218)

remove pandad from interfaces

maxime-desroches committed:
more codespell errors (#33223)

more spell

sshane committed:
make_can_msg: returns tuple (#33222)

* bump

* make_man_msg returns tuple

* fix CI.update typing

* bump

* better name

* Revert "better name"

This reverts commit 4deb38d4ed99e43721960f69da1dd46a1069eb42.

* common

maxime-desroches committed:
op.sh: lint (#33217)

* redo

* better

* fix this

* clean everything

* all files

* test

* debug

* get info

* revert

* only good files

* allow skip

* also this

* help section

maxime-desroches committed:
static_analysis: use 24.04 for everyone (#33227)

* noble

* test it

* Revert "test it"

This reverts commit 393868bf5fd14a0afaa6d2b05f632c76f9b52213.

maxime-desroches committed:
op lint: specify tests to run (#33229)

args

sshane committed:
selfdrive/car: rm make_can_msg (#33231)

* rm make_can_msg

* fix

sshane committed:
CAN fingerprinting: count empty CAN packets as frames (#33232)

* content of can packet won't affect can fingerprint time

* remove get_one_can

* fix comment

* actually we still want to be able to send empty can (xx fingerprint func does this if it runs out of msgs)

* fix process replay

sshane committed:
selfdrive/car: generic CAN send/receive callbacks (#33215)

* start

* hmm API doesn't seem great

* better

* rm line

* sendcan -> can_send callable for best static type coverage, list -> tuple msg

TODO: logcan

* more sendcan -> can_send references

* remove pandad's capnp from selfdrive/car

* fix and remove cereal from test_can_fingerprint

* test_fw_fingerprint: remove pandad, less cereal

* comment done too

* better comment

* fix test_models test

* niceeee it works

* move to can_definitions

* can't come up with a better name :(

* I think we can remove SimpleNamespace soon

* fix test_can_fingerprint.py

* maintain previous behavior

* Revert "maintain previous behavior"

This reverts commit f848fd32132391692c6191a305bb38f74091ec91.

* can test comment

* no need for get_one_can now!

* big clean up: no SimpleNamespace

* now self explanatory!

* not needed

* use empty can again since this is now real

* cmt

bongbui321 committed:
tools/logreader: add desc arg to run_across_segments() (#33226)

deanlee committed:
pandad: move getenv check to global scope (#33235)

move getenv check to global

bongbui321 committed:
tools/rerun: Rerun 0.18 (#33188)

maxime-desroches committed:
op.sh: fix venv on zsh + bash 3.2 (#33238)

* i love zsh

* check

maxime-desroches committed:
webcam: remove other cv2 usage (#33236)

av

maxime-desroches committed:
linux: setup panda + jungle udev rules (#33241)

* add rules

* allow failure on docker

adeebshihadeh committed:
Update CONTRIBUTING.md

adeebshihadeh committed:
tici: new updater build (#33246)

Co-authored-by: Comma Device <device@comma.ai>

bongbui321 committed:
tools/sim: resolve import error on MacOS running Metadrive (#33244)

fix dynamic lib linking on mac

maxime-desroches committed:
setup.sh: get full uname (#33248)

log

sshane committed:
Honda: round minEnableSpeed correctly (#33250)

round to even

maxime-desroches committed:
op.sh: consistent naming (#33249)

name

sshane committed:
move capnp CarParams decoding out of car_helpers (#33251)

* move capnp byte decoding out of car_helpers

* always wanted to type this

* and process replay

* this does nothing

* might as well

deanlee committed:
spinner: fix missing colon in QProgressBar border style (#33258)

fix border style

Quantizr committed:
Add testing closet to LogReader source (#33242)

* add testing closet to logreader source

* check source availability

* use internal_source_available

adeebshihadeh committed:
no more gh discussions

sshane committed:
Fix some car typing (#33256)

* ??

* fix these

* more typing and some fixes

* fix

* rm this for now

* fix

sshane committed:
Revert "spinner: fix missing colon in QProgressBar border style" (#33263)

Revert "spinner: fix missing colon in QProgressBar border style (#33258)"

This reverts commit 85d8d0eddd91baabb6bbef59a9d1f8463b70dc90.

sshane committed:
move the body events to card (#33262)

* move the body events to card

* rm import

sshane committed:
Update ui_preview.yaml

sshane committed:
remove map from ui preview (#33265)

* rm map from ui preview

* fix

sshane committed:
carState: add invalid sensors field (#33266)

* add to CS

* add vehicleSensorsInvalid to CarState

* fix ord

* clean up

commaci-public committed:
[bot] Update Python packages (#33272)

Update Python packages

Co-authored-by: Vehicle Researcher <user@comma.ai>

commaci-public committed:
[bot] Bump submodules (#33271)

bump submodules

Co-authored-by: Vehicle Researcher <user@comma.ai>

maxime-desroches committed:
ci: merge python packages and python submodules bump (#33274)

merge packages and python submodules

sshane committed:
[bot] Fingerprints: add missing FW versions from new users (#33276)

Export fingerprints

sshane committed:
fix fingerprint debug scripts (#33277)

* fix fp debugging

* fix these

maxime-desroches committed:
metadrive: use main branch (#33278)

use main branch

maxime-desroches committed:
op.sh: separate op.sh output from the real command  (#33282)

border

sshane committed:
interfaces: assume CarState is instantiated (#33284)

test to make sure mock doesn't crash!

sshane committed:
Move events out of car interfaces (#33280)

* chrysler down

* honda and ford

* we can pass the class for this

* do the rest

* clean that up

* remove EventName

* fix CI

* move smallest bodies to top

* rm todo

* eps_timer_soft_disable_alert is too ingrained in carcontroller

* re-did everything w/ no diff (except just passing cereal instead of class)

maxime-desroches committed:
op.sh: remove python version check (#33288)

python check

fredyshox committed:
controlsd: use livePose (#33283)

* Pose calibrator

* Fix static analysis

* Fix static

* Fix test_latcontrol

* Fix test_latcontrol

* Update services in process replay

* Fix static

* Matmul not mul

* Add assertion

* Move pose calibration to data_sample

* Update ref commit

* Remove llk from cycle alerts

* Deprecated nogps event

* Switch power_draw to lp

* Bring back noGps alert

* Add handling code back

* get_bool

* Bring inputsok back

sshane committed:
Car interface clean up (#33290)

* remove some defs

* now carstate

* remove carControl from CI update function

* almost merged that

* and that

maxime-desroches committed:
fix bash + python3 shebangs (#33287)

fix shebangs

sshane committed:
fix static analysis

maxime-desroches committed:
lint.sh: check shebang format for bash + python3 (#33286)

lint shebang

sshane committed:
Consistent spacing in car interfaces (#33293)

* formatting

* fix these

* not these

sshane committed:
Move buttonEvents to CarState (#33292)

* move mazda button events to carstate

* do  more

* remove

* some more

* clean up

sshane committed:
card: set mock carState fields (#33294)

* mock carstate inside card

* not used

sshane committed:
Common CI._update function (#33289)

* use CP

* no car control, consistent _update function signatures

* eh it's fine to name it whatever

* clean up

* oops

* !!

* now we can delete this!

* nobody does anymore

maxime-desroches committed:
lint.sh: better shebang lint (#33296)

-e

maxime-desroches committed:
retry uv with cache (#33297)

try uv with cache

sshane committed:
rm tesla (#33300)

* rm tesla

* more rm

* ugh we should remove dynamic imports soon

maxime-desroches committed:
replace python with python3 (#33303)

3

fredyshox committed:
ui: remove GPS status from the sidebar (#33304)

* Remove usage of llk in ui

* Remove GPS from sidebar

* Remove it from test

maxime-desroches committed:
op.sh: categorize commands (#33305)

names

sshane committed:
Copy kalman filter to selfdrive/car (#33307)

move to car/common/ and give proper names

sshane committed:
move live fingerprint debugging code to debug/car (#33308)

* remove cereal from fw_versions

* fix it

* do ecu_addrs and vin

* do last one

* move executable-ness

* notexe

* nice test

sshane committed:
body: remove controls import (#33309)

* body uses car's PIDController

* use a simple version

sshane committed:
docs: remove common import (#33310)

* add opcar

* fix references

* actually better to split this so opendbc will test docs conventions, and openpilot will test diffing

* not exe

* clean up

maxime-desroches committed:
setup.sh: fix macOS path specification (#33311)

realpath and mac

sshane committed:
GM: remove basedir import (#33312)

* remove these exceptions

* whoop

* ahhh

adeebshihadeh committed:
jenkins: remove unused pcStage

sshane committed:
run_process_on_route.py: use zstd

sshane committed:
toyota: pycapnp enum which should be called

sshane committed:
hyundai: declare type override of cruise_buttons

sshane committed:
hyundai: test some feature detection (#33315)

test some feature detection

sshane committed:
radard: remove redundant NoneType check

sshane committed:
fw_versions.py: mark raw in f-string

sshane committed:
card: fix return signature for update_events

sshane committed:
ecu_addrs.py: fix function argument typing

sshane committed:
test_fw_fingerprint.py: fix RAM HD test

adeebshihadeh committed:
rerun: update to 0.18 (#33316)

fredyshox committed:
paramsd: remove usages of llk nested structures (#33317)

* Remove usages of llk nested structures

* Rename to debugFitlerState

* Semicolon

* Initial parameters debugFilterState

sshane committed:
selfdrive/car: ban cereal and capnp (#33208)

* ban cereal and msgq

* common too

* do toyota/values.py

* do all fingerprints

* example without builder

* this still works, but no type checking anymore

* stash

* wtf, how does this work

* okay actually not bad

* safe

* epic!

* stash data_structures.py

* some clean up

* hell yeah

* clean up old file

* add to delete

* delete

This reverts commit 90239b7797ace31ee647a2fdbd67e0c3faa98dcf.

* switch more CarParams stuff over

remove unused

* fix car tests by removing cereal! mypy forgets about dataclass if we wrap it :(

* fix this too

* fix this too

* remove more cereal and add some good hyundai tests

* bunch more typing

* override default with 20hz radar

* temp capnp converter helper

* more lateralTuning

* small union replicator is better than what i was trying, and fixes mypy dynamic typing issues

* can keep all this the same now!

* type ret: CarParams, add more missing structs, revert lateralTuning changes (smaller diff!)

* revert more

* get first enum automatically, but ofc mypy doesn't pick up the new metaclass so can't use :(

would have been `CarParams.NetworkLocation()`

* Revert "get first enum automatically, but ofc mypy doesn't pick up the new metaclass so can't use :("

This reverts commit bb28b228becba932052d2fc5a4389784027435b1.

* remove cereal from car_helpers (TODO: caching)

* remove a bunch of temp lines

* use dataclass_transform!

* remove some car.CarParams from the interfaces

* remove rest of car.CarParams from the interfaces

* same which() API

* sort

* from cereal/cache from fingerprinting!

* more typing

* dataclass to capnp helper for CarParams, cached it since it's kinda slow

* (partial) fix process replay fingerprintig for new API

* latcontrollers take capnp

* forgot this

* fix test_models

* fix unit tests

* not here

* VehicleModel and controller still takes capnp CP since they get it from Params()

* fix modeld test

* more fix

* need to namespace to structs, since CarState is both class and struct

* this was never in the base class?!

* clean that up again

* fix import error

fix import error

* cmts and more structs

* remove some more cereal from toyota + convert CarState to capnp

* bruh this was wrong

* replace more cereal

* EventName is one of the last things...

* replace a bunch more cereal.car

* missing imports

* more

* can fix this typing now

* proper toyota+others CS typing!

* mypy can detect return type of CS.update() now

* fix redeclaration of cruise_buttons type

* mypy is only complaining about events now

* temp fix

* add carControl struct

* replace CarControl

i hope there's no circular imports in hyundai's CC

* fine now

* lol this was wrong too

* fix crash

* include my failed attempts at recursively converting to dataclass (doesn't implicitly convert types/recursively :( )

but attrs does, maybe will switch in the future

* clean up

* try out attr.s for its converter (doesn't work recursively yet, but interesting!)

* Revert "try out attr.s for its converter (doesn't work recursively yet, but interesting!)"

This reverts commit ff2434f7bbd45a4d4bfb21f7d6712d1f1c3bcde9.

* test processes doesn't fail anymore (on toyota)!

* fix honda crash

* stash

* Revert "stash"

This reverts commit c1762af4e776790e4ad1322ad4ce0610157346e0.

* remove a bunch more cereal!

* LET'S GOOO

* fix these tests

* and these

* and that

* stash, something is wrong with hyundai enable

* Revert "stash, something is wrong with hyundai enable"

This reverts commit 39cf327def258e2959fe23cd7a550a858f6d8f03.

* forgot these

* remove cereal from fw_versions

* Revert "remove cereal from fw_versions"

This reverts commit 232b37cd409b55d04b1afc90d4a80c49e710eb56.

* remove rest of the cereal exceptions!

* fix that

* add typing to radard since I didn't realize RI.update() switched from cereal to structs

* and here too!

* add TODO for slots

* needed CS to be capnp, fix comparisons, and type hint car_specific so it's easier to catch type issues (capnp isn't detected by mypy :( )

* remove the struct converter

* save ~4-5% CPU at 100hz, we don't modify after so no need to deepcopy

btw pickle.loads(pickle.dumps()) is faster by ~1% CPU

* deepcopy -> copy: we can technically make a reference, but copy is almost free and less error-prone

saves ~1% CPU

* add non-copying asdict function

* should save ~3% CPU (still 4% above baseline)

* fix that, no dict support

* ~27% decrease in time for 20k iterations on 3X (3.37857 -> 2.4821s)

* give a better name

* fix

* dont support none, capitalize

* sheesh, this called type() on every field

* remove CS.events, clean up

* bump card %

* this was a bug on master!

* add a which enum

* default to pid

* revert

* update refs

* not needed, but consistent

* just Ecu

* don't need to do this in this pr

* clean up

* no cast

* consistent typing

* rm

* fix

* can do this if we're desperate for the last few %

* Revert "can do this if we're desperate for the last few %"

This reverts commit 18e11ac7883a0a56583750b1cc5a2b13011e7299.

* type this

* don't need to convert carControl

* i guess don't support set either

* fix CP type hint

* simplify that

sshane committed:
car: move conversions.py into common/

sshane committed:
remove import linter (#33319)

* rm import linter, we're done!

* revert

* uv lock

* Revert "uv lock"

This reverts commit 5e46f48e8a5738ea5121da6f2d1c7210dcefe053.

sshane committed:
move selfdrive/car to opendbc (#32630)

* move most of /car

* move some car tests

move some car tests

* fix selfdrive/car/tests

* fix selfdrive/controls tests

* fix the rest of the selfdrive tests

* bump opendbc

* fix all tests

* few more non-test references

* remove opcar and move docs to car

fix these debugging scripts

fix docs

* bump opendbc and panda

forgot panda

samehmohamed88 committed:
Cleaning up README for tools directory (#33322)

cleaning up readme

Co-authored-by: Sameh <Sameh Mohamed>

andiradulescu committed:
agnos updater: set decompress max_length (#33320)

* agnos: decompress max_length

* flash last chunk after eof

* don't decompress more than length

* cleanup

---------

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

adeebshihadeh committed:
update test

sshane committed:
card: fix memory leak from nested function scoping (#33328)

* stash

* no other leaks! pm.send grows memory usage by ~20mb but that's it

* undo

* clean up

AlexandreSato committed:
Update what-is-a-car-port.md (#33323)

sshane committed:
Remove car labeler (#33329)

* can?

* car still makes sense

* bump

* opendbc

* ?

* bump

* revert

sshane committed:
fix labeler syntax

commaci-public committed:
[bot] Update Python packages (#33327)

* Update Python packages

* fix opendbc

---------

Co-authored-by: Vehicle Researcher <user@comma.ai>
Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

adeebshihadeh committed:
research -> autonomy

deanlee committed:
cereal/SubMaster: encapsulate frequency management in FrequencyTracker (#33252)

* encapsulate frequency management

* apply reviews

* early return, avoiding unnecessary calculations

* simplify avg freq calc

deanlee committed:
SubMaster: add self.services attribute for consistent service management (#33268)

simplify service handing

sshane committed:
update car docs in weekly repo maintenance (#33345)

* does this work?

* fix

* fix

* what

commaci-public committed:
[bot] Update Python packages (#33347)

Update Python packages

Co-authored-by: Vehicle Researcher <user@comma.ai>

sshane committed:
move test_lateral_limits.py to opendbc

sshane committed:
ui: fix wrong path color indexing (#33350)

* fix

* clean up

* clean up

* clean up

sshane committed:
move some car interface tests into opendbc (#33352)

* bump

* this is in opendbc

* so is radar interface

* clean up

* bump

* can import from opendbc

* bump

deanlee committed:
drive_helpers.py: iterate button_timers with items() (#33351)

iterate button_timers with items()

deanlee committed:
scons: remove Wno-error=unused-but-set-variable (#33355)

remove Wno-error=unused-but-set-variable

deanlee committed:
scons:  remove -Wno-deprecated-register and -Wno-register  (#33356)

remove flag

bongbui321 committed:
rerun: adapt to the change in release of 0.18 (#33359)

deanlee committed:
scons: remove QtMultimedia lib and include path from build configuration (#33358)

remove multimedia

adeebshihadeh committed:
bump panda

fredyshox committed:
timed: use gps location packet for time (#33306)

* GPS helpers

* Use GPS socket in timed

* Use helper in controlsd

* Use updated and logMonoTime

* Helper type annot

* Not updated

* Increase time threashold

deanlee committed:
scons: remove duplicate linker flags for -ljson11, -lzmq  (#33361)

fix duplicate link

deanlee committed:
cereal: remove unused shared library creation (#33363)

clear cereal/SConscript

deanlee committed:
scons: only enable swaglog for submodules (#33365)

only enable swaglog for submodules

deanlee committed:
ui: remove QtQml and related dependencies (#33357)

* remove QtQml

* remove qtdeclarative5-dev

maxime-desroches committed:
add back pre-commit hook (#33366)

* add pre-commit

* space

* post

* e

adeebshihadeh committed:
ui: move more builds behind --minimal (#33368)

adeebshihadeh committed:
bump panda

adeebshihadeh committed:
these have nothing to do with openpilot (#33369)

deanlee committed:
scons: cleanup cpppath (#33370)

cleanup cpppath

adeebshihadeh committed:
third_party: setup raylib (#33367)

* setup raylib

* x86 build

* add larch

* darwin

---------

Co-authored-by: Comma Device <device@comma.ai>

deanlee committed:
ui: add prefix PRIME_TYPE_ to PrimeType enum values (#33371)

Add Prefix PRIME_TYPE_ to PrimeType Enum Values

commaci-public committed:
[bot] Update Python packages (#33377)

Update Python packages

Co-authored-by: Vehicle Researcher <user@comma.ai>

maxime-desroches committed:
Increase fetch depth for submodules (#33378)

4000

sshane committed:
car helpers file (#33379)

* helpers file

* clean up

bongbui321 committed:
sim: fix code formatting (#33381)

sim: fix code

adeebshihadeh committed:
openpilot isn't required for a car port PR (#33383)

* disable car docs diff

* mv routes.py

* bump opendbc

deanlee committed:
ui: fix memory corruption by using std::vector for PubMaster initialization (#33390)

Fix memory Corruption Issue

adeebshihadeh committed:
manager: remove LastUpdateTime default

adeebshihadeh committed:
no PC anymore

adeebshihadeh committed:
ubuntu: remove extra packages

maxime-desroches committed:
git rewrite (#33392)

maxime-desroches committed:
Update rewrite.sh

sshane committed:
bump msgq

maxime-desroches committed:
git rewrite: backup old repo (#33393)

backup

deanlee committed:
ci: add onroad alerts to ui report (#33394)

* add onroad alerts to ui report

* print

adeebshihadeh committed:
bump rednose

maxime-desroches committed:
git rewrite: ignore release branches

adeebshihadeh committed:
cleanup duplicate serial docs

deanlee committed:
ci: add all onroad hud elements to ui report (#33395)

* add model

* merge master

* make static analysis happy

deanlee committed:
bridge: implement MSGQ to ZMQ bridge with subscriber-based publishing (#32862)

implement MSGQ to ZMQ bridge with subscriber-based publishing

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

deanlee committed:
ci: add cameras to ui reports (#33401)

add cameras

deanlee committed:
ui: remove deprecated Qt ui_snapshot (#33402)

remove ui_snapshot

deanlee committed:
ui: fix Incorrect path in .gitignore for test_ui/report (#33403)

fix gitignore

deanlee committed:
bridge: drain each socket up to a fixed number (#33400)

drain each socket up to a fixed number

mitchellgoffpc committed:
Add gas/brake press probs to ModelDataV2 (#33382)

* Add gas/brake press probs to ModelDataV2

* Updated model_replay_ref_commit

fredyshox committed:
azure_container: pass the filename explicitely when using get_url (#33407)

* Pass filename explicitely when using azure container

* Do not update segments in test_processes

* openpilotci_source_zst

* Fix static analysisc

fredyshox committed:
Deprecate liveLocationKalman (#33405)

* Remove usages of llk

* Deprecate it

* Add scenarios back

* Fix orientationNED valid

* Regenerate

* Increase tolerances and update segment

* Remove calibration

royjr committed:
ui: update arabic translations (#33411)

Update main_ar.ts

AlexandreSato committed:
small typo on run.py (#33413)

sshane committed:
joystickd: non-linear response function

sshane committed:
joystickd: lower deadzone

deanlee committed:
ci: add driver camera to ui report (#33409)

add driver camera

maxime-desroches committed:
reduce metadrive size (#33391)

* minimal

* uv.lock

* remove anisotropic filtering

* new wheel

sshane committed:
joystick: perform deadzone first

sshane committed:
joystickd: works with game controller on device (#33414)

* state what we support

* more

* wording

AlexandreSato committed:
Bot comment only for first timers (#33412)

* Wellcome comment just for first timers

* Update auto_pr_review.yaml

https://github.com/actions/first-interaction/blob/main/README.md

adeebshihadeh committed:
move PID controller to common/ (#33419)

adeebshihadeh committed:
ui: gc experiments

adeebshihadeh committed:
controlsd: deprecate alertBlinkingRate (#33422)

* controlsd: deprecate alertBlinkingRate

* simplify

* update refs

adeebshihadeh committed:
controlsd: split out state/alerts into selfdriveState (#33421)

* setup selfdriveState

* little more

* update refs

* migration

* all too slow

deanlee committed:
ci: add controlsState back to ui report (#33424)

add controlsState back

royjr committed:
ui: show personality updates (#33406)

* personality-alerts

* Update events.py

* Update car.capnp

* no AudibleAlert

* shorter

* no VisualAlert

* one liner

* smaller

* normal

* what am I doing wrong?

* it works, but is it messy?

* arg all

* lil more

* update that

---------

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

adeebshihadeh committed:
update test_alerts

deanlee committed:
ci: display lead car indicator in on-road screenshots (#33416)

Show lead car in on-road screenshots

deanlee committed:
replay: migrate old ControlState event to new format during log file parsing (#33425)

Migrate Old ControlState Event to New Format During Log File Parsing

deanlee committed:
cabana: update event Handling from CONTROLS_STATE to SELFDRIVE_STATE (#33426)

check selfdrive_state

deanlee committed:
ci: integrate offroad alerts and update notifications to ui report (#33408)

add offroad alerts

deanlee committed:
ui: remove copy initialization of dialogs (#33427)

remove copy initialization of dialogs

adeebshihadeh committed:
athena: use dongle ID on /persist/ when available (#33259)

* athena: use dongle ID on /persist/ when available

* comment

* test

* cleanup

deanlee committed:
ci: add body to ui report (#33429)

add body

adeebshihadeh committed:
carEvent -> onroadEvent (#33432)

deanlee committed:
ui: refactor driver monitor updating and rendering into `DriverMonitorRenderer` class (#33375)

* Refactor Driver Monitor Updating and Rendering

add comments

* rebase master

* rename dmon to driver_monitoring

AlexandreSato committed:
Fix small typos on run.py (#33434)

deanlee committed:
remove to common and no default return value

maxime-desroches commented:
This PR was closed because of a git history rewrite.
Please read https://github.com/commaai/openpilot/issues/33399 for what to do with your fork and your PRs.
